src/client/app/components/CompareLineControlsComponent.tsx

```diff
@@ -0,0 +1,134 @@
+import { dateRangeToTimeInterval, timeIntervalToDateRange } from '../utils/dateRangeCompatibility';
+import { selectSelectedLanguage } from '../redux/slices/appStateSlice';
+import { Value } from '@wojtekmaj/react-daterange-picker/dist/cjs/shared/types';
+import * as moment from 'moment';
+import { TimeInterval } from '../../../common/TimeInterval';
+
+/**
+ * @returns compare line control component for compare line graph page
+ */
+export default function CompareLineControlsComponent() {
+	const dispatch = useAppDispatch();
+	const shiftAmount = useAppSelector(selectShiftAmount);
+	const timeInterval = useAppSelector(selectQueryTimeInterval);
+	const locale = useAppSelector(selectSelectedLanguage);
+	const shiftInterval = useAppSelector(selectShiftTimeInterval);
+
+	// Hold value of shifting option (week, month, year, or custom)
+	const [shiftOption, setShiftOption] = React.useState<ShiftAmount>(shiftAmount);
+	// Hold value to store the custom date range for the shift interval
+	const [customDateRange, setCustomDateRange] = React.useState<TimeInterval>(shiftInterval);
+
+	// Update the shift interval when the shift option changes
+	React.useEffect(() => {
+		if (shiftOption !== ShiftAmount.custom && timeInterval.getIsBounded()) {
+			const { shiftedStart, shiftedEnd } = shiftDate(timeInterval.getStartTimestamp(), timeInterval.getEndTimestamp(), shiftOption);
+			const newInterval = new TimeInterval(shiftedStart, shiftedEnd);
+			dispatch(updateShiftTimeInterval(newInterval));
+		}
+	}, [shiftOption, timeInterval]);
+
+	// Update custom date range value when shift interval changes
+	React.useEffect(() => {
+		setCustomDateRange(shiftInterval);
+	}, [shiftInterval]);
+
+	// Handle changes in shift option (week, month, year, or custom)
+	const handleShiftOptionChange = (value: string) => {
+		if (value === 'custom') {
+			setShiftOption(ShiftAmount.custom);
+			dispatch(updateShiftAmount(ShiftAmount.custom));
+		} else {
+			const newShiftOption = value as ShiftAmount;
+			setShiftOption(newShiftOption);
+			dispatch(updateShiftAmount(newShiftOption));
+		}
+	};
+
+	// Update date when the data range picker is used in custome shifting option
+	const handleCustomShiftDateChange = (value: Value) => {
+		setCustomDateRange(dateRangeToTimeInterval(value));
+		dispatch(updateShiftTimeInterval(dateRangeToTimeInterval(value)));
+	};
+
+	return (
+		<>
+			<div key='side-options'>
+				<p style={{ fontWeight: 'bold', margin: 0 }}>
+					<FormattedMessage id='shift.date.interval' />
+					{/* <TooltipMarkerComponent helpTextId='help.shift.date.interval' /> // TODO: Add later */}
+				</p>
+				<Input
+					id='shiftDateInput'
+					name='shiftDateInput'
+					type='select'
+					value={shiftOption}
+					invalid={shiftAmount === ShiftAmount.none}
+					onChange={e => handleShiftOptionChange(e.target.value)}
+				>
+					<option value="none" hidden disabled>{translate('select.shift.amount')}</option>
+					<option value="day">{translate('1.day')}</option>
```

Comment:
Maybe look at src/client/app/components/IntervalControlsComponent.tsx for how it loops over object to do this. I think in other places it might loop over an enum. Would that make sense here or is it too specific to this case?